controllers/fluxcd/application-controller.go

```diff
@@ -0,0 +1,426 @@
+}
+
+func (r *ApplicationReconciler) reconcileKustomization(app *v1alpha1.Application) (err error) {
+	ctx := context.Background()
+
+	fluxApp := app.Spec.FluxApp.DeepCopy()
+	appNS, appName := app.GetNamespace(), app.GetName()
+
+	kusList := createBareFluxKustomizationListObject()
+	if err = r.List(ctx, kusList, client.InNamespace(appNS), client.MatchingLabels{
+		"app.kubernetes.io/managed-by": getKusManagerName(appNS, appName),
+	}); err != nil {
+		return
+	}
+
+	kusMap := make(map[string]*unstructured.Unstructured, len(kusList.Items))
+	for _, kus := range kusList.Items {
+		kusMap[kus.GetName()] = kus.DeepCopy()
+	}
+
+	kusNum := len(app.Spec.FluxApp.Spec.Config.Kustomization)
+	for i := 0; i < kusNum; i++ {
+		kusDeploy := fluxApp.Spec.Config.Kustomization[i]
+		kusName := getKustomizationName(kusDeploy.Destination.TargetNamespace)
+
+		if kus, ok := kusMap[kusName]; !ok {
+			// not found
+			// create
+			err = r.createKustomization(ctx, app, kusDeploy)
+		} else {
+			// found
+			// update this kus
+			err = r.updateKustomization(ctx, kus, app, kusDeploy)
+		}
+	}
+	return
+}
+
+func (r *ApplicationReconciler) createKustomization(ctx context.Context, app *v1alpha1.Application, deploy *v1alpha1.KustomizationSpec) (err error) {
+	appNS, appName := app.GetNamespace(), app.GetName()
+	kusNS, kusName := appNS, getKustomizationName(deploy.Destination.TargetNamespace)
+
+	kus := createBareFluxKustomizationObject()
+	setFluxKustomizationFields(kus, app, deploy)
+	kus.SetNamespace(kusNS)
+	kus.SetName(kusName)
+	kus.SetOwnerReferences([]metav1.OwnerReference{
+		{
+			APIVersion: "gitops.kubesphere.io/v1alpha1",
+			Kind:       "Application",
+			Name:       appName,
+			UID:        app.GetUID(),
+		},
+	})
+	kus.SetLabels(map[string]string{
+		"app.kubernetes.io/managed-by": getKusManagerName(appNS, appName),
+	})
+	err = r.Create(ctx, kus)
+	return
+}
+
+func (r *ApplicationReconciler) updateKustomization(ctx context.Context, kus *unstructured.Unstructured, app *v1alpha1.Application, deploy *v1alpha1.KustomizationSpec) (err error) {
+	setFluxKustomizationFields(kus, app, deploy)
+	err = r.Update(ctx, kus)
+	return
+}
+
+func setFluxKustomizationFields(kus *unstructured.Unstructured, app *v1alpha1.Application, deploy *v1alpha1.KustomizationSpec) {
+	_ = argocd.SetNestedField(kus.Object, app.Spec.FluxApp.Spec.Source.SourceRef, "spec", "sourceRef")
+
+	configs, _ := argocd.InterfaceToMap(*deploy)
+	for k, config := range configs {
+		if k != "destination" {
+			_ = unstructured.SetNestedField(kus.Object, config, "spec", k)
+		}
+	}
+	_ = argocd.SetNestedField(kus.Object, deploy.Destination.KubeConfig, "spec", "kubeConfig")
+	_ = unstructured.SetNestedField(kus.Object, deploy.Destination.TargetNamespace, "spec", "targetNamespace")
+}
+
+func isHelmOrKustomize(app *v1alpha1.Application) AppType {
+	if app.Spec.FluxApp.Spec.Config.HelmRelease != nil {
+		return HelmRelease
+	}
```

Comment:
Need check if `Kustomization` not equal to nil ? and then return an error ?

Reply:
Yes. I'll fix it.